src/middlewared/middlewared/plugins/system.py

```diff
@@ -1358,12 +1365,35 @@ def set_language(self):
     def set_crash_reporting(self):
         CrashReporting.enabled_in_settings = self.middleware.call_sync('system.general.config')['crash_reporting']
 
+# Update Birthday Date
+async def _update_birthday(middleware):
+    # Sync clock
+    middleware.logger.debug('Synchornization the clock for system birthday')
+    birthday = None
+    timeout=60
+
+    while birthday == None:
+        birthday = sync_clock()
```

Comment:
`sync_clock` performs I/O, you can't do this in asynchronous methods. I propose to move it to system service as private method so you can take advantage of I/O thread pool we have.